src/common/node/childProcess.ts

```diff
@@ -55,44 +61,32 @@ export class ChildProcess {
         return this.exec(command, options).outcome.then(stdout => stdout.toString());
     }
 
-    public spawnWithExitHandler(command: string, args?: string[], options: ISpawnOptions = {}): ISpawnResult {
+    public spawn(info: ISpawnInfo, options: ISpawnOptions = {}): ISpawnResult {
         let outcome = Q.defer<void>();
-        let commandWithArgs = command + " " + args.join(" ");
+        let commandWithArgs = info.command + " " + info.args.join(" ");
 
-        let spawnedProcess = child_process.spawn(command, args, options);
+        let spawnedProcess = child_process.spawn(info.command, info.args, options);
         spawnedProcess.once("error", (error: any) => {
             outcome.reject(error);
         });
-        spawnedProcess.once("exit", (code: number) => {
-            if (code === 0) {
-                outcome.resolve(void 0);
-            } else {
-                outcome.reject(ErrorHelper.getInternalError(InternalErrorCode.CommandFailedWithErrorCode, commandWithArgs, code));
-            }
-        });
 
-        return {
-              spawnedProcess: spawnedProcess,
-              stdin: spawnedProcess.stdin,
-              stdout: spawnedProcess.stdout,
-              stderr: spawnedProcess.stderr,
-              outcome: outcome.promise };
-    }
-
-    public spawn(command: string, args?: string[], options: ISpawnOptions = {}): ISpawnResult {
-        let outcome = Q.defer<void>();
-        let commandWithArgs = command + " " + args.join(" ");
-        let spawnedProcess = child_process.spawn(command, args, options);
-        spawnedProcess.once("error", (error: any) => {
-            outcome.reject(ErrorHelper.getNestedError(error, InternalErrorCode.CommandFailed, commandWithArgs));
-        });
+        if (info.waitForExit) {
+            spawnedProcess.once("exit", (code: number) => {
+                if (code === 0) {
+                    outcome.resolve(void 0);
+                } else {
+                    outcome.reject(ErrorHelper.getInternalError(InternalErrorCode.CommandFailed, commandWithArgs, code));
+                }
+            });
+        } else {
+            Q.delay(300).done(() => outcome.resolve(void 0));
```

Comment:
The problem with "turn spawn into a promise friendly thing" is what exactly does that promise do? Is it just `return Q(child_process.spawn)`? If that's the case, then it's not really worth promisifying. Promises are for asynchronous things, and the current informal spec here is "return a promise which is rejected if we fail to launch it (~= the process asynchronously emits an error event within the first 300ms) and resolved otherwise on the assumption that the process was successfully started".